### PR TITLE
feat(github-repository-search): add filter preferences

### LINKED ISSUE #10367: [GitHub Repository Search] Search only within your repositories
### Extension

https://www.raycast.com/thomas/github-repository-search

### Description

I'll add some background here. I developed long time ago with a colleague a chrome extension. Code is here: https://github.com/firstandthird/github-repo-search

This extension is great since once you're in a couple of orgs, you get a ton of repositories and it's hard to comb through them. The downside is that is mainly for chrome, the UX on FF is subpar and doesn't work nicely on ARC as the omnibar doesn't seem to work either.

Right now this raycast extension extension (and the official one) searches through all of GH. My request is to add an option that just searches through your own repositories.

### Who will benefit from this feature?

I assume a lot of people? I don't think the GH search is useful by itself when it comes to combing through your repos

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by mheidinger**
## Description

This adds additional preferences to the GitHub repository search to filter the search for users/organisations, to include forks or any other static GitHub query.

Resolves #10367.

## Screencast

https://github.com/raycast/extensions/assets/8899636/7d0db163-ceb5-432a-ab84-896dfcd55461

https://github.com/raycast/extensions/assets/8899636/52f30dfe-9cc8-425f-99b9-b3d89e78d215

https://github.com/raycast/extensions/assets/8899636/93c57057-1fc8-4384-8ec0-7aeb23389c36

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/github-repository-search/package.json [hunk 1]
```diff
@@ -29,6 +29,27 @@
       "required": false,
       "title": "GitHub Base URL",
       "description": "Optional Base URL for GitHub Enterprise (e.g. https://github.acme-inc.com/api/v3). Defaults to the public GitHub."
+    },
+    {
+      "name": "searchInOrgs",
+      "type": "textfield",
+      "required": false,
+      "title": "Organizations / Users to search in",
+      "description": "Comma separated list of organizations or users to search in. Leave empty to search in all repositories. Use @me to search in your own repositories."
```
Comment (thomaspaulmann) [inline extensions/github-repository-search/package.json:38]: I assume this limts the search to only the users/orgs? Maybe it's good to clarify this in the description.
```suggestion
      "description": "Limit the search so a comma-separated list of users or organizations, e.g. 'mheidinger, raycast'. Tip: Use '@me' to search only your own repositories."
```

File: extensions/github-repository-search/package.json [hunk 2]
```diff
@@ -29,6 +29,27 @@
...
+      "required": false,
+      "label": "Include Forks",
+      "description": "Include forked repositories in search results."
+    },
+    {
+      "name": "additionalQuery",
+      "type": "textfield",
+      "required": false,
+      "title": "Additional Query",
+      "description": "Additional static query to filter repositories (e.g. language:typescript)."
```
Comment (thomaspaulmann) [inline extensions/github-repository-search/package.json:52]:
```suggestion
      "description": "Additional filters to limit the repositories search, e.g. 'language:typescript'. See https://ray.so/0XgqB6q for all filters."
```

File: extensions/github-repository-search/src/useRepositories.ts [hunk 3]
```diff
@@ -35,15 +36,45 @@ query SearchRepositories($searchText: String!) {
...
+  }
+
+  const searchTextParts: string[] = [];
+  if (preferenceValues.searchInOrgs) {
+    const orgs = preferenceValues.searchInOrgs
+      .split(",")
+      .map((org) => org.trim())
+      .filter((org) => org.length > 0);
+    if (orgs.length > 0) {
+      const orgsSearchText = orgs.map((org) => `user:${org}`).join(" ");
```
Comment (thomaspaulmann) [inline extensions/github-repository-search/src/useRepositories.ts:51]: Does this actually work for organizations like `raycast`? The docs mention that there is a `user:` and `org:` filter.

**Review by thomaspaulmann — CHANGES_REQUESTED** [review]
This is really cool and thanks for the addition. I left a few comments around naming for the preferences. Plus a question if we need to differentiate users and organizations in teh search query syntax.

File: extensions/github-repository-search/package.json [hunk 4]
```diff
@@ -29,6 +29,27 @@
       "required": false,
       "title": "GitHub Base URL",
       "description": "Optional Base URL for GitHub Enterprise (e.g. https://github.acme-inc.com/api/v3). Defaults to the public GitHub."
+    },
+    {
+      "name": "searchInOrgs",
```
[the remark this answers is not in the document]
Comment (mheidinger) [reply extensions/github-repository-search/package.json]: Sure, renamed it.

File: extensions/github-repository-search/package.json (hunk 1 above)
Comment (mheidinger) [reply extensions/github-repository-search/package.json]: Took over the suggestion

File: extensions/github-repository-search/src/useRepositories.ts (hunk 3 above)
Comment (mheidinger) [reply extensions/github-repository-search/src/useRepositories.ts]: Yes, you can see that in the second video of the PR description, where `raycast,@me` is set and `includeForks` is enabled. There it returns results for both raycast and my personal repos.

From what I understood from testing, filtering for `user:X` and `org:X` does behave identical with the exception that `org:@me` does not work.

**Comment by mheidinger** [comment]
While running the extension locally I noticed warnings about reserved keyboard shortcuts:
- Open Wiki (cmd+w)
- Open Pull Requests (cmd+p)

Any proposal to what these could be changed? I first changed the one for PRs to cmd+r but now realized that this is already taken by "Browse Releases". cmd+shift+p is also already taken...

**Comment by pernielsentikaer** [comment]
What about cmd+shift+(o/p)

File: extensions/github-repository-search/src/useRepositories.ts (hunk 3 above)
Comment (thomaspaulmann) [reply extensions/github-repository-search/src/useRepositories.ts]: Ok, then lets leave as is. If people enter `user:thomaspaulmann` and `org:raycast` as additional filters, it should still work.

**Comment by thomaspaulmann** [comment]
> What about cmd+shift+(o/p)

I think those shortcuts could work. We could also do `⌘` `⇧` `W` for Wiki if we want to stick with the first letter. Leaving this to you @mheidinger. Do you want to update the shortcuts in this PR?

**Comment by pernielsentikaer** [comment]
That could also work. I just know that `⌘` `⇧` `W` is always close window, so my brain doesn't work with that but I don't use the hotkey myself 🙂

**Comment by mheidinger** [comment]
As its a tiny change, I'd just include it in here. Changed them in a separate commit :)

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋

Looks good to me, approved 🔥

**Pull request merged**